File: 2022B/b.py
import math


def solve():
    n = int(input())
    count = 0
    for i in range(1, int(math.sqrt(n))+1):
        if (n % i == 0):
            if ((n // i) == i):
                if (str(i) == str(i)[::-1]):
                    count += 1
            else:
                if (str(i) == str(i)[::-1]):
                    count += 1
                if (str((n//i)) == str((n//i))[::-1]):
                    count += 1
    print(count)


if __name__ == "__main__":
    T = int(input())
    for i in range(1, T+1):
        print(f'Case #{i}: ', end="")
        solve()

File: 2022B/b.cpp
#include <bits/stdc++.h>
using namespace std;
bool isPalindrome(string S)
{
    string P = S;
    reverse(P.begin(), P.end());
    if (S == P)
        return true;
    return false;
}
void solve()
{
    int n, count = 0;
    cin >> n;
    for (int i = 1; i <= sqrt(n); i++)
    {
        if (n % i == 0)
        {
            cout << i << " ";
            if (n / i == i)
            {
                if (isPalindrome(to_string(i)))
                    count++;
            }
            else
            {
                if (isPalindrome(to_string(i)))
                    count++;
                if (isPalindrome(to_string((n / i))))
                    count++;
            }
        }
    }
    // cout << count << "\n";
}

int main(void)
{
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    int T;
    cin >> T;
    for (int i = 1; i <= T; i++)
    {
        cout << "Case #" << i << ": ";
        solve();
    }
    return 0;
}